### ISSUE TITLE
AllowMissingVariables = false should report missing variables at any depth.

### ISSUE THREAD

**Issue by flowchartsman** [cbroglie/mustache#12]
given the following template

```
You have {{count}} items:
{{#sections}}
--{{name}}--
{{#items}}
{{.}}
{{/items}}
{{/sections}}
```

If I do the following:

``` go
    mustache.AllowMissingVariables = false
    s, err := mustache.Render(tmpl, map[string]interface{}{
        //"count": 2,
        "sections": map[string]interface{}{
            "name":  "Stuff",
            "items": []string{"foo", "bar"},
        },
    })
    fmt.Println(s, err)
```

I receive `Missing variable "count"`

Like I expect.  The same goes for `sections`.  But, if I give it the following:

``` go
    mustache.AllowMissingVariables = false
    s, err := mustache.Render(tmpl, map[string]interface{}{
        "count": 2,
        "sections": map[string]interface{}{
            //"name": "Stuff",
            "items": []string{"foo", "bar"},
        },
    })
    fmt.Println(s, err)
```

I get:

```
You have 2 items:
----
foo
bar
```

I would expect something along the lines of `Missing variable "sections.name"` or, at the very least `Missing variable "name"`.

I feel like, if you're going to have a global variable called "AllowMissingVariables", it should apply at every level or, at the very least, the limits of its scope should be clearly noted in the docs.

**Comment by cbroglie** [comment]
I agree this is a valid issue. Not sure when I will have a chance to address it, PRs are welcome :smile:

**Comment by cbroglie** [comment]
Fixed by #39
